flake.nix

```diff
@@ -14,31 +14,63 @@
         let
           pkgs = import nixpkgs {
             inherit system;
-            overlays = [ ocaml-overlay.overlay ];
+            overlays = [ ocaml-overlay.overlays."${system}".default ];
           };
-          inherit (pkgs) lib;
-          myPkgs = pkgs.recurseIntoAttrs (import ./nix {
-            inherit pkgs;
-            doCheck = true;
-          }).native;
-          myDrvs = lib.filterAttrs (_: value: lib.isDerivation value) myPkgs;
+          pollinate =
+            (pkgs.callPackage ./nix { doCheck = true; });
         in {
-          devShell = (pkgs.mkShell {
-            inputsFrom = lib.attrValues myDrvs;
-            buildInputs = with pkgs;
-              with ocamlPackages; [
-                ocaml-lsp
-                ocamlformat
-                odoc
-                ocaml
-                dune_3
-                nixfmt
-              ];
-          });
+          devShells = {
+            default = (pkgs.mkShell {
+              inputsFrom = [ pollinate ];
+              buildInputs = with pkgs;
+                with ocamlPackages; [
+                  ocaml-lsp
+                  ocamlformat
+                  odoc
+                  ocaml
+                  dune_3
+                  nixfmt
+                ];
+            });
+          };
 
-          defaultPackage = myPkgs.pollinate;
+          packages = pkgs.lib.mapAttrs (version: ocamlPackages:
+            (pkgs.callPackage ./nix {
+              inherit ocamlPackages;
+              doCheck = true;
+            })) (builtins.removeAttrs pkgs.ocaml-ng [
+              "overrideDerivation"
```

Comment:
Does it mean "we want to remove everything except ocaml 4.13"?
Isn't there any more beautiful way to do it?

Reply:
Yes, we should invert it